src/components/Search/SearchFiltersParticipantsSelector.tsx

```diff
@@ -0,0 +1,223 @@
+    const personalDetails = usePersonalDetails();
+    const {options, areOptionsInitialized} = useOptionsList();
+
+    const [betas] = useOnyx(ONYXKEYS.BETAS);
+    const [isSearchingForReports] = useOnyx(ONYXKEYS.IS_SEARCHING_FOR_REPORTS, {initWithStoredValues: false});
+    const [selectedOptions, setSelectedOptions] = useState<OptionData[]>([]);
+    const [searchTerm, debouncedSearchTerm, setSearchTerm] = useDebouncedState('');
+    const cleanSearchTerm = useMemo(() => searchTerm.trim().toLowerCase(), [searchTerm]);
+
+    const defaultOptions = useMemo(() => {
+        if (!areOptionsInitialized) {
+            return defaultListOptions;
+        }
+
+        return OptionsListUtils.getFilteredOptions(
+            options.reports,
+            options.personalDetails,
+            betas,
+            '',
+            selectedOptions,
+            CONST.EXPENSIFY_EMAILS,
+            false,
+            true,
+            false,
+            {},
+            [],
+            false,
+            {},
+            [],
+            true,
+            false,
+            false,
+            0,
+            undefined,
+            true,
+        );
+    }, [areOptionsInitialized, betas, options.personalDetails, options.reports, selectedOptions]);
+
+    const chatOptions = useMemo(() => {
+        return OptionsListUtils.filterOptions(defaultOptions, cleanSearchTerm, {
+            betas,
+            selectedOptions,
+            excludeLogins: CONST.EXPENSIFY_EMAILS,
+            maxRecentReportsToShow: CONST.IOU.MAX_RECENT_REPORTS_TO_SHOW,
+        });
+    }, [defaultOptions, cleanSearchTerm, betas, selectedOptions]);
+
+    const sections = useMemo(() => {
+        const newSections: OptionsListUtils.CategorySection[] = [];
+        if (!areOptionsInitialized) {
+            return newSections;
+        }
+
+        const formattedResults = OptionsListUtils.formatSectionsFromSearchTerm(
+            cleanSearchTerm,
+            selectedOptions,
+            chatOptions.recentReports,
+            chatOptions.personalDetails,
+            personalDetails,
+            true,
+        );
+
+        const isCurrentUserSelected = selectedOptions.find((option) => option.accountID === chatOptions.currentUserOption?.accountID);
+
+        newSections.push(formattedResults.section);
+
+        if (chatOptions.currentUserOption && !isCurrentUserSelected) {
+            newSections.push({
+                title: '',
+                data: [chatOptions.currentUserOption],
+                shouldShow: true,
+            });
+        }
+
+        newSections.push({
+            title: '',
+            data: chatOptions.recentReports,
+            shouldShow: chatOptions.recentReports.length > 0,
+        });
+
+        newSections.push({
+            title: '',
+            data: chatOptions.personalDetails,
+            shouldShow: chatOptions.personalDetails.length > 0,
+        });
+
+        return newSections;
```

Comment:
Can't do that. There might be empty sections inserted, because code above this is written like this: 
```
        newSections.push({
            title: '',
            data: chatOptions.recentReports,
            shouldShow: chatOptions.recentReports.length > 0,
        });
```

so might be there is no results, but still there are empty section with `{ data: [] }`.

I will count the header inside memo like you suggested, but will do a different check.